File: starters/Java/bomberjam/Logger.java
package bomberjam;

import java.io.Closeable;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;

public class Logger implements Closeable {
    private PrintWriter file;

    public Logger() {
        this.file = null;
    }

    public void setup(String filename) throws IOException {
        if (this.file == null)
            this.file = new PrintWriter(new FileWriter(filename, false));
    }

    public void debug(String text) {
        this.write("DEBUG", text);
    }

    public void info(String text) {
        this.write("INFO", text);
    }

    public void warn(String text) {
        this.write("WARN", text);
    }

    public void error(String text) {
        this.write("ERROR", text);
    }

    private void write(String level, String text) {
        if (this.file != null) {
            this.file.println(level + ": " + text);
            this.file.flush();
        }
    }

    public void close() {
        if (this.file != null) {
            this.file.close();
            this.file = null;
        }
    }
}

File: starters/CSharp/Bomberjam/Logger.cs
using System;
using System.IO;

namespace MyBot.Bomberjam
{
    public class Logger : IDisposable
    {
        private StreamWriter _file;

        public Logger()
        {
            this._file = null;
        }

        public void Setup(string filename)
        {
            if (this._file == null)
                this._file = new StreamWriter(File.OpenWrite(filename));
        }

        public void Debug(string text)
        {
            this.Write("DEBUG", text);
        }

        public void Info(string text)
        {
            this.Write("INFO", text);
        }

        public void Warn(string text)
        {
            this.Write("WARN", text);
        }

        public void Error(string text)
        {
            this.Write("ERROR", text);
        }

        private void Write(string level, string text)
        {
            if (this._file != null)
            {
                this._file.WriteLine(level + ": " + text);
                this._file.Flush();
            }
        }

        public void Close()
        {
            if (this._file != null)
            {
                this._file.Close();
                this._file = null;
            }
        }

        public void Dispose()
        {
            this.Close();
        }
    }
}